**Contributors:** salaamridwan@yahoo.com, ajisegiriblessing@gmail.com, suprisebaloyi17@gmail.com, 
                      mercyfrancis18@gmail.com

**Video link:** https://www.linkedin.com/posts/sewedo-ajisegiri-6372a6218_cancerstudy-biomarkerdiscovery-hackbiointern-activity-7238229392684965888-f8FT?utm_source=share&utm_medium=member_desktop

**Team Lead:** Surprise Baloyi

**Members:** Sewedo Ajisegiri, Mercy Francis, Salaam Ridwan. 

**Comprehensive Molecular Characterization of Gastric Adenocarcinoma**

**Introduction**

Before 2014, scientists and the World Health Organization (WHO) worked to classify gastric cancer to guide urgency priority for patient therapy. However, the classic classification based on tissue types and location had little to no significant clinical function. Moreover, the increasing molecular profiling of deadly cancer necessitates the need to provide a more encompassing classification of gastric cancer that takes into cognizance the key drivers. In this study, gene expression analysis was used to identify crucial genes associated with various types of gastric carcinoma. 

**Methodology**

A total of 295 gastric adenocarcinoma tissue samples were collected in a fresh frozen state, with no prior chemotherapy or radiation treatment, and with informed consent and approval from the local Institutional Review Board. Gene expression analysis, one of the six different molecular techniques used, measured gene activity across the expression platforms within the study. 

**Results** 

This study identified four primary genomic subtypes of gastric cancer: Epstein–Barr virus-positive tumors, microsatellite unstable tumors, and those with chromosomal instability as shown in Fig. 1 


![Fig  1](https://github.com/user-attachments/assets/32904b0c-3337-4f4b-b000-ce6c43fe0718)







**Figure 1:** A representation of the genomic subtypes of gastric carcinoma with sample size data**.**

By analyzing the mRNA sequence data for alternative splicing events, about 30% of the sample data showed skipping Mesenchymal Epithelial Transition (MET 2\) exon. This has been associated with increased MET expression. Also, novel variants of the same gene, namely MET exon 18 and 19, were skipped, as shown in Fig. 2\. 




![Figure 2](https://github.com/user-attachments/assets/833fc53f-1d51-434d-a45d-66e467b4155d)





**Figure 2:** Reads Per Kilobase per Million mapped reads (RPKM) expression of MET in samples with and without exon (A) or exon 18 and/or 19 (B) skipping in MET. 

Based on Reverse Phase Protein Array (RPPA) data analysis, 45 proteins were also expressed or phosphorylated across the four molecular subtypes. Notably, EGFR was significantly phosphorylated in the CIN subtype while expression of the p53 gene was abnormal, which was consistent with TP53 mutation in the same subtype, as shown in Fig. 3\. 




![Fig  3](https://github.com/user-attachments/assets/b9d252d0-e8fc-4e13-9846-67d2a80fe0a0)






**Figure 3:** Unsupervised hierarchical clustering of RPPA data, identifying three clusters, including reactive, invasive, and proliferative.

**Discussion**

Clearly, gene expression analysis plays a crucial role in identifying and classifying gastric cancer types with significant clinical functions. This is because the over-expressed EGFR, P53, and MET are not only biomarkers but potential therapeutic genes. 

**Conclusion**

Based on the findings of this study, gene expression analysis is crucial for cancer classification, biomarker discovery, and therapeutic protein identification.  

**Reference**

The Cancer Genome Atlas Research Network. (2014). Comprehensive molecular characterization of gastric adenocarcinoma. *Nature*, *513*(7517), 202–209. https://doi.org/10.1038/nature13480

