# Hackbio-Internship

This repository houses the tasks assigned to me for the duration of my internship at Hackbio.

## Repository Structure

- [Stage 0](#stage-0)
- [Stage 1](#stage-1)
- [Stage 2](#stage-2)
- [Stage 3](#stage-3)
- [Stage 4](#stage-4)



## Stage 0

[Details for Stage 1, if any]

## Stage 1

[Details for Stage 2, if any]

## Stage 2

### Task Overview

In this stage, we were expected to visualize and interpret a gene expression dataset to generate a heatmap and perform downstream functional enrichment analysis. The aim was to help us understand and learn how to interpret patterns of gene expression and the biological significance of differentially expressed genes. The task involved data preprocessing, visualization, and interpretation of functional enrichment results.

### Key Steps

1. Download and use one of the provided datasets.
2. Generate heatmaps with different color palettes and clustering options.
3. Subset genes based on significant up/down-regulation.
4. Perform functional enrichment analysis.
5. Create a visualization of the top 10 pathways.
6. Describe the top 3 enriched pathways according to biological process.

## Stage 3

### Task Overview

In this stage, we were asked to identify potential cancer biomarkers from a given dataset using differential expression and machine learning models. This project required creative thinking and collaboration, leveraging the strengths of both fields.

### Key Steps

1. Select a cancer type/subtype and download data from TCGA.
2. Clean and preprocess the data.
3. Conduct differential expression and functional enrichment analyses.
4. Prepare the data for machine learning.
5. Perform feature selection and classification (kNN or random forest).
6. Write a comprehensive report on the findings.

## Stage 4

### Task Overview

In this stage, we were asked to reproduce the unsupervised clustering of gliomas (n = 516 LGG) based on methylation levels to classify the six IDH statuses as reported in a specified paper. The goal was to demonstrate our ability to perform unsupervised clustering using methylation data and to evaluate the clustering performance.

### Key Steps

1. Read and understand the specified paper.
2. Obtain and preprocess the dataset.
3. Implement the analysis pipeline (kNN and random forest for ML; DEA for Biomarker-Devs).
4. Generate a final report comparing results with the target paper.

## How to Use This Repository



## Contact

[Mercys' Gmail](mercyfrancis18@gmail.com)
